### ISSUE TITLE
not nice with http_proxy or https_proxy variables.

### ISSUE THREAD

**Issue by equwal** [miquels/davfs3#22]
I use privoxy for my proxy.
```
echo "$http_proxy"
127.0.0.1:3988
echo "$https_proxy"
127.0.0.1:3988
```
```
 mount -t webdavfs -ousername=user,password=pass "https://us0.co/sabredav/server.php" /home/user/mnt


mount.webdavfs: Options "https://us0.co/sabredav/server.php/": proxyconnect tcp: net/http: TLS handshake timeout
```
it times out
setting the variables to `""` is a workaround.

**Comment by miquels** [comment]
This is not a bug in webdavfs, this is the behaviour of the Go http library. If `http_proxy` or `https_proxy` is set to `localhost` or `127.0.0.1` or `::1`, then the proxy setting is _ignored_. This is hardcoded behaviour, it cannot be changed. A workaround is to put an entry in /etc/hosts that resolves to 127.0.0.1, for example

```
127.0.0.1   localproxy
```
.. and then use `localproxy` in `http_proxy` and `https_proxy`.

**Comment by equwal** [comment]
Indeed, it seems that a bunch of my go programs have been buggy in this way. Thanks.
